src/MauiIcons.Core/Base/BaseMauiIcon.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace MauiIcons.Core.Base;
+public abstract class BaseMauiIcon : ContentView
```

Comment:
In MAUI the `CompressedLayout` has not been ported from Xamarin.
Every `View` we use represents a concrete view on the native layer, sometimes it is even mapped to two views (wrapper + content).
Considering this is an `Icon` which aims to be used potentially multiple times in each page, I would consider inheriting from `View` and then implement the `ILabel` interface explicitly.
We can then reuse MAUI label handlers to render this component.

Probably this is an improvement for a future PR, but still I wanted to express my performance concerns.

Reply:
Oh, that's super helpful. I should probably need to look for into native layer and makes sense. I'll consider architecting with this concern in mind.